staging/src/k8s.io/client-go/examples/create-update-delete-deployment/main.go

```diff
@@ -53,13 +53,13 @@ func main() {
 		panic(err)
 	}
 
-	deploymentsClient := clientset.AppsV1beta1().Deployments(apiv1.NamespaceDefault)
+	deploymentsClient := clientset.AppsV1().Deployments(apiv1.NamespaceDefault)
 
-	deployment := &appsv1beta1.Deployment{
+	deployment := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "demo-deployment",
 		},
-		Spec: appsv1beta1.DeploymentSpec{
+		Spec: appsv1.DeploymentSpec{
 			Replicas: int32Ptr(2),
 			Template: apiv1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
```

Comment:
`spec.Selector` is a required value in `apps/v1` deployments. It will need to match `spec.template.metadata.labels`. So you need to include the `Selector` field in the `DeploymentSpec`.

`apps/v1beta1`: https://github.com/kubernetes/api/blob/master/apps/v1beta1/types.go#L316

`apps/v1`: https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L272